File: ios/Sources/Mia21/Models/BotModels.swift
//
//  BotModels.swift
//  Mia21
//
//  Created on November 4, 2025.
//  Copyright © 2025 Mia21. All rights reserved.
//
//  Description:
//  Bot-related data models.
//  Includes bot information and creation requests.
//

import Foundation

// MARK: - Bot

public struct Bot: Codable, Identifiable {
  public let botId: String
  public let name: String
  public let prompt: String
  public let llmIdentifier: String
  public let temperature: Double
  public let maxTokens: Int
  public let language: String
  public let voiceId: String?
  public let isDefault: Bool
  public let customerId: String
  public let createdAt: String?
  public let updatedAt: String?
  
  // Computed property for Identifiable conformance
  public var id: String {
    return botId
  }
  
  // Note: No custom CodingKeys needed - APIClient uses .convertFromSnakeCase
  // snake_case from API (e.g., "bot_id") automatically maps to camelCase (e.g., botId)
}

// MARK: - Bots Response

public struct BotsResponse: Codable {
  public let bots: [Bot]
  public let count: Int
  
  // Note: No custom CodingKeys needed - APIClient uses .convertFromSnakeCase
}

// MARK: - Bot Create Request

public struct BotCreateRequest {
  public let botId: String
  public let name: String
  public let voiceId: String
  public let additionalPrompt: String?
  public let isDefault: Bool?

  public init(
    botId: String,
    name: String,
    voiceId: String,
    additionalPrompt: String? = nil,
    isDefault: Bool? = false
  ) {
    self.botId = botId
    self.name = name
    self.voiceId = voiceId
    self.additionalPrompt = additionalPrompt
    self.isDefault = isDefault
  }
}

File: android/lib/src/main/kotlin/com/mia21/models/BotModels.kt
/**
 * BotModels.kt
 * Mia21
 *
 * Created on November 25, 2025.
 * Copyright © 2025 Mia21. All rights reserved.
 *
 * Description:
 * Bot-related data models.
 * Includes bot information and creation requests.
 */

package com.mia21.models

import kotlinx.serialization.SerialName
import kotlinx.serialization.Serializable

/**
 * Represents an AI bot/assistant
 */
@Serializable
data class Bot(
    @SerialName("bot_id")
    val botId: String,
    val name: String,
    val prompt: String,
    @SerialName("llm_identifier")
    val llmIdentifier: String,
    val temperature: Double,
    @SerialName("max_tokens")
    val maxTokens: Int,
    val language: String,
    @SerialName("voice_id")
    val voiceId: String? = null,
    @SerialName("is_default")
    val isDefault: Boolean,
    @SerialName("customer_id")
    val customerId: String,
    @SerialName("created_at")
    val createdAt: String? = null,
    @SerialName("updated_at")
    val updatedAt: String? = null
)

/**
 * Response containing a list of bots
 */
@Serializable
data class BotsResponse(
    val bots: List<Bot>,
    val count: Int
)

/**
 * Request to create a new bot
 */
data class BotCreateRequest(
    val botId: String,
    val name: String,
    val voiceId: String,
    val additionalPrompt: String? = null,
    val isDefault: Boolean? = false
)

